Analytics-CSharp/Segment/Analytics/Timeline.cs

```diff
@@ -161,10 +161,18 @@ internal void Add(Plugin plugin)
             analytics.AnalyticsScope.Launch(analytics.AnalyticsDispatcher, async () =>
             {
                 Settings? settings = await plugin.Analytics.SettingsAsync();
+
+                // This is basically never null, look at the comment in SettingsAsync
+                if (settings == null)
+                {
+                    return;
+                }
+
+                // Fetch system afterwards for a minuscule but cool performance gain
                 System system = await analytics.Store.CurrentState<System>();
-                // Don't initialize unless we have updated settings from the web.
-                // CheckSettings will initialize everything added before then, so wait until other inits have happened.
-                if (settings.HasValue && system._initializedPlugins.Count > 0)
+
+                // Check for nullability because CurrentState returns default(IState) which could make the .Count throw a NullReferenceException
+                if (system._initializedPlugins != null && system._initializedPlugins.Count > 0)
```

Comment:
Not quite. You have a constructor with 4 parameters, and an optionable HashSet parameter. When returning a default(struct), it targets the default constructor. To my understanding, since the parameters are not available when creating a default instance, it cannot call it - hence you get an "empty" object.

In this case, a solution also could be to just add

`internal System() { _initializedPlugins = new(); }`

And it would avoid this check.